src/ophys_etl/workflows/pipeline_module.py

```diff
@@ -170,6 +194,13 @@ def input_args_path(self) -> Path:
         )
         return args_path
 
+    def validate_input_args(self) -> None:
+        """Validates module input args after it's been processed by
+        EnhancedJSONEncoder"""
+        encoded_json = json.dumps(self.inputs, cls=EnhancedJSONEncoder)
```

Comment:
I'm not sure why this step is necessary. Why can't `self.module_schema.load(data=self.inputs)` be called instead?

Reply:
The inputs need to be processed by the JSONEncoder first for them to be representative of the actual arguments that get passed into the module.

Reply:
Please give a concrete example. It should work without it.

Reply:
Well, if you think about what the JSONEncoder does, it converts Path objects to str and it converts enum objects to str. 
The schemas don't deal with Path objects or enum objects, they require str.

Reply:
ok makes sense. Could you please add a comment why are you are encoding using `EnhancedJSONEncoder`?